tests/rptest/utils/node_metrics.py

```diff
@@ -21,3 +26,10 @@ def disk_free_bytes(self) -> list[float]:
 
     def disk_space_alert(self) -> list[float]:
         return self._get_metrics_vals("storage_disk_free_space_alert")
+
+    def wait_until_populated(self, timeout_sec=15):
```

Comment:
What is the definition of `ready` though? IMHO, this is a symptom of something we should fix; I think having health data available for all nodes in the cluster-level (i.e. controller) quorum should be an invariant of being in that quorum. I have a hard time reasoning about the various split brain scenarios that could cause health data to be partially filled in (populated) otherwise.

Reply:
> What is the definition of ready though?

that the NodeMetrics object is ready to be used

> having health data available for all nodes in the cluster-level (i.e. controller) quorum should be an invariant

the original version of my comment was that you could move this wait_until barrier into redpanda.py so that it was a pre-condition of the node being considered to have been started. but that is a slippery slope and before long we'll have a lot of conditions piling up.

we should probably consider delaying the start of some services like the kafka layer, metrics, admin api until we've done a minimum amount of bootstrapping on startup. i think tha tis probably what you are suggesting.

Reply:
On re-read.. I think I missed your point. The name shouldn't have to change if we fix the implementation detail. So, take my thoughts on simplifying health monitor availability as an aside. I'll update the name here and resolve this soon.

Reply:
oof.. Laggy GH comments today. I reloaded a moment ago and did not see your most recent comment here :shrug: Yeah associating cluster quorum (top-level, think minimal controller state) with invariants like metadata being available is a great simplifying assumption.

Reply:
This was addressed w/ the suggested rename.